docs/pages/07_service.adoc

```diff
@@ -0,0 +1,955 @@
+
+*Permission graph*
+
+// https://en.wikipedia.org/wiki/Graph-based_access_control
+
+JUDO's security mechanism is based on a logical permission graph, where the entry point for external users to the application is the actor. Calling the services granted to the actor the user can access transfer objects. Once the user has access to a transfer object, they can call the services granted by the transfer object (and can access further transfer objects).
+
+The primary benefit of this mechanism is that it provides a direct mapping between the user's permissions and the transfer objects and services they can access. This makes it easier to manage permissions and control access to the application.
+
+The example below shows how a transfer object can grant services. Users who received the PersonTransfer object can call the delete service.  This service can only be called by the users who have access to the PersonTransfer object.
+
+*Syntax:*
+
+----
+transfer <name> (<entity> <alias>) [grants <export>, ...] [{
+    [member] ...
+}]
+----
+
+*Example:*
+
+----
+transfer PersonTransfer(Person person) grants PersonExport {
+    field String email = salesPerson.email ;
+}
+
+export PersonExport(Person person) {
+    service void delete() {
+        // implement this body in Java
+    }
+}
+----
+
+The complete authorization mechanism can be defined by carefully provisioning services to actors and transfer objects. Always be aware of that if a transfer object has already been received by an external user, then they are authorized to call the services provided by the transfer object.
+
+=== Guard
+
+A guard is a boolean expression that must evaluate to true in order to continue the execution.
+
+Guards can be defined for
+
+* services,
+* exports and
+* actors.
+
+To add a guard to services use `guard` as follows:
+
+*Example:*
+
+----
+export PersonExport(Person person) {
+    service void delete() guard person.leads!size() == 0;
+}
+----
+
+The example above defines a guard, and the delete service is not executed if that person has a lead.
+
+// TODO: what is the error message?
+
+To add a guard to exports use `guard` as follows:
+
+*Example:*
+
+----
+export PersonExport(Person person) guard person.leads!size() == 0 {
+    service void delete() ;
+    service Lead[] addLead(Lead lead) ;
+}
+----
+
+The example above defines a guard for export, and the services are not executed if that person has a lead.
+
+To add a guard to actors use `guard` as follows:
+
+*Example:*
+
+----
+actor SalesPersonActor(SalesPerson salesPerson)
+    realm "COMPANY"
+    identity salesPerson.email
+    guard salesPerson.leads!size() > 0;
+----
+
+The example above defines a guard for export, and the salesperson can only log in if he has at least one lead.
```

Comment:
This is not true. If a person exists in the IDM system (openid / keycloak) then they will always be able to log in. The only thing an actor guard can do is facilitate `authorization` = denies access to e.g. services if the condition is not met (403 response).

Reply:
I agree with @noherczeg. Moreover, this is the similar case the grants. My advice is the actor need to have only relam and identity clause.